src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -80,6 +81,9 @@ public Person toModelType() throws IllegalValueException {
         }
         final Email modelEmail = new Email(email);
 
```

Comment:
So i think the null check for the isDone String should be here

Reply:
I think when we make isDone attribute have a nullcheck, i think it will affect the add/ edit command. Making it a non null would mean that the isDone attribute is a compulsory field. Might need to check

Reply:
> I think when we make isDone attribute have a nullcheck, i think it will affect the add/ edit command. Making it a non null would mean that the isDone attribute is a compulsory field. Might need to check

But isDone is a compulsory field. It just so happens that it is auto assumed as false when someone uses the add command. But what if when u try to import data instead, and some people dont have a done field? In that case we need to be able to check and assign the default value false right?

Reply:
so i was thinking about sth along the lines of:

`IsDone modelIsDone;`
`if (isDone == null) {
     modelIsDone = new IsDone('false');
}`
`if (!(IsDone.isValidIsDone(isDone))) {
            throw new IllegalValueException(IsDone.MESSAGE_CONSTRAINTS);
        }`

basically like how all the other fields are checked. The only difference is that in the case that isDone is null, we dont throw a 'MissingFieldFormat' but instead just assume it to be false

Reply:
> I still think there wont be a situation where an isDone attribute of a valid person will be null.

If the storage files are not able to be manually edited by the user then yes i would agree with you, since our functions only add valid persons by design. But since users can edit the json file, and the CSV file we receive comes from a third party (at least in our scenario) we cant assume that this will always be the case. I mean thats why the checks for the other fields exist right? If everything is valid, why is there a need to check if phone number is valid, or email is valid?

Reply:
if it's null, the current approach is to just load in a new empty database if the json file are missing any attiribute. Are u implying we should update this approach?

Reply:
For csv, if the isDone attribute is missing, this will be the case where isDone.equals(""). This means that the person is still valid would be treated as false (will not throw any exeption either), as per your concern. In other cases, if they are missing(example name.equals(""), it will throw the expection and will not be a valid person. this exception is however caught (unlike the json counterpart) and will move on to find the next valid person

Reply:
The current approach is if the string isDone equals to "true" (or any variations in capitalizations) will make the isDone.value TRUE. Any other string value wold result in the isDone.value to be FALSE

Reply:
TLDR isDone = null is not needed because

JSON - if isDone is missing ( or if any missing attribute for the matter), this means the JSON file has been tempered and the whole database would not import anything. (this has been the default approach)

CSV - if Isdone is missing, the string would return as "". This person would still be valid and be treated as isDone.Value FALSE. If name/phone/email is missing, this person will be considered invalid and csv will look for the next VALID person

Reply:
> JSON - if isDone is missing ( or if any missing attribute for the matter), this means the JSON file has been tempered and the whole database would not import anything. (this has been the default approach)

If we stick to the original approach, then you should still throw a MISSING_FIELD_MESSAGE_FORMAT. So still need to check if its null.

Reply:
Hi, i did as you said (checking if isDone is null) and here is what i got.

![Capture4](https://user-images.githubusercontent.com/77221700/135740721-28d749d6-3137-46b8-b3bd-c96c7a6c70fb.PNG)

Here is the reason why I think this is so:

The reason why other fields throw a missing field message format is due to converting a person to a jsonAdaptedPerson.
![Capture1](https://user-images.githubusercontent.com/77221700/135740513-a8cc750f-753b-45cc-bcb6-280074634c70.PNG)

This happens when there is a command issued (edit/add/save etc)

As I followed ur previous suggestion of changing isDone to boolean, 

![Capture2](https://user-images.githubusercontent.com/77221700/135740587-2dd93a18-59a7-441c-8730-28f3722f97b4.PNG)

This means that isDone.value will not be null as it's of boolean datatype. The string inside JSON will not be null due to this as well

![Capture3](https://user-images.githubusercontent.com/77221700/135740653-9cedc722-986a-416d-805f-5b56163642e2.PNG)

This is possibly why the string isDone in the jsonAdaptedPerson will never be null

Reply:
> The reason why other fields throw a missing field message format is due to converting a person to a jsonAdaptedPerson.

I think its bcos of line 38? If u kept the original line 38 `this.isDone = isDone`, then isDone can be null.

Reply:
I tried checking with other fields as well

![image](https://user-images.githubusercontent.com/77191381/135741153-18b4a3f4-5e24-4870-8e35-63572dce29cf.png)

![image](https://user-images.githubusercontent.com/77191381/135741164-4a488de3-8605-484e-be1f-89f870b504af.png)

email can still be null

Reply:
> I think its bcos of line 38? If u kept the original line 38 `this.isDone = isDone`, then isDone can be null.

hmm. reverting back to the original would still result in isDone never being null. Even if we were to include this isDone = null, there still wont be anything thrown. What do u suggest we do? Do you want to work and test it out? Maybe it would be easier that way

Reply:
Never mind i found out why. It's the ordering of the checking. its null check BEFORE valid is done. resolved

Reply:
Null value in isDone will throw an exception message "MIssing FIELD" instead of "INVALID ISDONE"